### DISCUSSION TITLE
请问如果一个NB2里配置了两个适配器，这两个适配器可以同时处理消息吗？

### DISCUSSION THREAD

**Discussion by junnanjiang** [nonebot/discussions#267] [问答]
bot.py里配置了两个适配器：
```
#bot.py
driver.register_adapter(ONEBOT_V11Adapter)
driver.register_adapter(NTCHATAdapter)
```
然后go-cqhttp和[ntchat-client](https://github.com/JustUndertaker/ntchat-client)接收qq和wechat消息，
在log里也能看到NB2的确收到了qq和wechat发来的消息，
但是NB2只处理的go-cqhttp发来的消息，ntchat-client发来的消息并没有被处理，log如下：

**以下为NB2里接收到了go-cqhttp发来的消息，可以正常处理消息（收到//echo456q，返回456q）**
> 12-11 16:10:21 [INFO] nonebot_plugin_gocqhttp | [44449999] 收到好友 Run(33338888 ) 的消息: //echo456q (1694334643)
> 12-11 16:10:21 [SUCCESS] nonebot | OneBot V11 44449999| [message.private.friend]: Message 1694334643 from 33338888 "//echo456q"
> 12-11 16:10:21 [DEBUG] nonebot | Checking for matchers in priority 1...
> 12-11 16:10:21 [INFO] nonebot | Event will be handled by Matcher(type='message', module=nonebot.plugins.echo)
> 12-11 16:10:21 [DEBUG] nonebot | Running Matcher(type='message', module=nonebot.plugins.echo)
> 12-11 16:10:21 [DEBUG] nonebot | Running handler Dependent(call=handle_echo)
> 12-11 16:10:21 [DEBUG] nonebot | OneBot V11 | Calling API send_msg
> 12-11 16:10:21 [INFO] nonebot_plugin_gocqhttp | [44449999] 发送好友 33338888 (33338888 )  的消息: [{"type":  ... (-653415605)
> 12-11 16:10:21 [INFO] nonebot | Matcher(type='message', module=nonebot.plugins.echo) running complete
> 12-11 16:10:21 [DEBUG] nonebot | Checking for matchers in priority 10...
> 12-11 16:10:21 [DEBUG] nonebot | Checking for matchers in priority 999...

**以下为NB2里接收到了ntchat-client发来的消息，但没有被处理（//echo456rr应该返回456rr，实际没有返回）**
> 12-11 16:13:43 [INFO] nonebot | ntchat | Bot wxid_7e62t4v31ne322 connected
> 12-11 16:13:43 [INFO] logging | connection open
> 12-11 16:14:12 [SUCCESS] nonebot | ntchat wxid_7e62t4v31ne322 | [MT_RECV_TEXT_MSG]: Message 7566387533129168520 from Run: //echo456rr
> 12-11 16:14:12 [DEBUG] nonebot | Checking for matchers in priority 1...
> 12-11 16:14:12 [DEBUG] nonebot | Checking for matchers in priority 10...
> 12-11 16:14:12 [DEBUG] nonebot | Checking for matchers in priority 999...

**还有异常的地方是，用了chatgpt插件，收到ntchat-client的消息，是可以调用Handler的，但会被skipped（也是无法发送消息）**
> 12-11 15:50:17 [SUCCESS] nonebot | ntchat wxid_7e62t4v31ne322 | [MT_RECV_TEXT_MSG]: Message 6036453049901389914 from Run: //0728hello
> 12-11 15:50:17 [DEBUG] nonebot | Checking for matchers in priority 1...
> 12-11 15:50:17 [DEBUG] nonebot | Checking for matchers in priority 10...
> 12-11 15:50:17 [DEBUG] nonebot | Checking for matchers in priority 999...
> 12-11 15:50:17 [INFO] nonebot | Event will be handled by Matcher(type='message', module=nonebot_plugin_chatgpt)
> 12-11 15:50:17 [DEBUG] nonebot | Running Matcher(type='message', module=nonebot_plugin_chatgpt)
> 12-11 15:50:17 [DEBUG] nonebot | Running handler Dependent(call=ai_chat, parameterless=(Depends(Dependent(call=check_cooldown)),))
> 12-11 15:50:17 [DEBUG] nonebot | Handler Dependent(call=ai_chat, parameterless=(Depends(Dependent(call=check_cooldown)),)) **skipped**
> 12-11 15:50:17 [INFO] nonebot | Matcher(type='message', module=nonebot_plugin_chatgpt) running complete
> 12-11 15:50:17 [DEBUG] nonebot | Stop event propagation

就是想问下，如果NB2里同时用了这两个适配器，是不是无法同时处理消息？

**Comment by yanyongyu** [comment]
nb本身可以同时处理多个事件消息，第一个没有响应可能是因为`to_me`检查没通过，比如没有at或者适配器的问题。第二个skipped是因为插件指定了仅响应onebot适配器的消息，这个请向插件反馈问题。
